mobile-app/app/screens/AppNavigator/screens/Balances/screens/AddressBookScreen.tsx

```diff
@@ -46,7 +49,10 @@ export function AddressBookScreen ({ route, navigation }: Props): JSX.Element {
     if (hasPendingJob || hasPendingBroadcastJob) {
       return
     }
-    onAddressSelect(address)
+    // condition to make address component unclickable from settings page
+    if (address !== undefined && onAddressSelect !== undefined) {
```

Comment:
can we reuse the variable you created at the beginning?

Reply:
can't reuse since the logic is checking for different conditions